### PR TITLE
Add IndicGenBenchFlores Dataset

### PR DESCRIPTION
**Pull request by bp-high**
- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [ ] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: mteb/tasks/BitextMining/multilingual/IndicGenBenchFloresBitextMining.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    "gbm_Deva",
+    "gom_Deva",
+    "hne_Deva",
+    "raj_Deva",
+    "mai_Deva",
+    "mni_Mtei",
+    "mup_Deva",
+    "mwr_Deva",
+    "sat_Olck",
+]
```
Comment (KranthiGV) [inline mteb/tasks/BitextMining/multilingual/IndicGenBenchFloresBitextMining.py:41]: Inconsistency due to - vs _
Can you use `-` everywhere? AFAIK that's the correct way to represent.

Ref: `ISO_LANGUAGE_SCRIPT = str  # a 3-letter ISO 639-3 language code followed by a 4-letter ISO 15924 script code (e.g. "eng-Latn")`

**Review by KranthiGV — COMMENTED** [review]
Code changes LGTM

File: mteb/tasks/BitextMining/multilingual/IndicGenBenchFloresBitextMining.py (hunk 1 above)
Comment (bp-high) [reply mteb/tasks/BitextMining/multilingual/IndicGenBenchFloresBitextMining.py:41]: @KranthiGV Basically I copied this for ease but the ones getting passed to the Task Metadata are transformed using get_lang_pairs to be in format similar to tatoeba(https://github.com/embeddings-benchmark/mteb/blob/main/mteb/tasks/BitextMining/multilingual/TatoebaBitextMining.py) and a similar operation has also been performed in FloresBitext class (https://github.com/embeddings-benchmark/mteb/blob/main/mteb/tasks/BitextMining/multilingual/FloresBitextMining.py#L220)
Screenshot 2024-05-16 at 12 32 13 PM

Comment (isaac-chung) [reply mteb/tasks/BitextMining/multilingual/IndicGenBenchFloresBitextMining.py:41]: It's good to be consistent. I noticed that some existing datasets, such as `mteb/tasks/BitextMining/multilingual/FloresBitextMining.py` has underscores and uses `-` to signify the cross lingual component. @KennethEnevoldsen or @imenelydiaker what's the preferred way here?

Comment (bp-high) [reply mteb/tasks/BitextMining/multilingual/IndicGenBenchFloresBitextMining.py:41]: @isaac-chung btw the input to the eval_langs param of TaskMetadata is consistent and does not use underscores in all existing datasets the underscores are replaced by '-' and follow pattern similar to Tatoeba(dictionary mapping huggingface subsets to lists of languages), 
Screenshot 2024-05-16 at 2 38 44 PM

Screenshot 2024-05-16 at 2 37 50 PM

I have followed the same pattern.

Comment (isaac-chung) [reply mteb/tasks/BitextMining/multilingual/IndicGenBenchFloresBitextMining.py:41]: Oh gotcha. I had missed your earlier comment. Thanks!

**Review by isaac-chung — APPROVED** [review]
Looks good! Feel free to add points - 2 for the dataset and 4 for each new language not already covered in bitext mining tasks, and 2 for each reviewer.

**Comment by bp-high** [comment]
@isaac-chung . Add 26 points(2+ 4(mwr) + 4(mup)+4(raj) +4(gbm) +4(boy) +4(bgc) ) for dataset and 2 for each reviewer.

**Comment by bp-high** [comment]
@isaac-chung Resolved the merge conflicts.

**Pull request merged**